src/main/java/org/example/dao/RouteDAOImpl.java

```diff
@@ -0,0 +1,108 @@
+            statement.setLong(3, route.getShortestDistance());
+            statement.executeUpdate();
+
+            try (ResultSet generatedKeys = statement.getGeneratedKeys()) {
+                if (generatedKeys.next()) {
+                    int id = generatedKeys.getInt(1);
+                    route.setId(id);
+                    System.out.println("Route inserted successfully. Generated ID: " + id);
+                }
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void updateRoute(Route route) {
+        try (Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+             PreparedStatement statement = connection.prepareStatement("UPDATE routes SET shortestDistance = ? WHERE startPointId = ? AND endPointId = ?")) {
+            statement.setLong(1, route.getShortestDistance());
+            statement.setInt(2, route.getStartPoint().getId());
+            statement.setInt(3, route.getEndPoint().getId());
+            int rowsAffected = statement.executeUpdate();
+            if (rowsAffected > 0) {
+                System.out.println("Route updated successfully.");
+            } else {
+                System.out.println("Route not found.");
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public Route getRoute(int id) {
+        try (Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+             PreparedStatement statement = connection.prepareStatement("SELECT startPointId as r.startPoint, endPointId as r.endPoint, shortestDistance as r.distance FROM routes r WHERE id = ?")) {
+            statement.setInt(1, id);
+            try (ResultSet resultSet = statement.executeQuery()) {
+                if (resultSet.next()) {
+                    int startPointId = resultSet.getInt("startPointId");
+                    int endPointId = resultSet.getInt("endPointId");
+                    long shortestDistance = resultSet.getLong("shortestDistance");
+
+                    PointDAO pointDAO = new PointDAOImpl();
+                    Point startPoint = pointDAO.getPoint(startPointId);
+                    Point endPoint = pointDAO.getPoint(endPointId);
+
+                    Route route = new Route(startPoint, endPoint, shortestDistance);
+                    route.setId(id);
+                    return route;
+                }
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    @Override
+    public List<Route> getRoutes() {
+        List<Route> routes = new ArrayList<>();
+        try (Connection connection = CONNECTIONPOOL.getConnectionFromPool();
+             PreparedStatement statement = connection.prepareStatement("SELECT id as r.id ,startPointId as r.startPoint, endPointId as r.endPoint, shortestDistance as r.distance FROM routes r");
```

Comment:
This one could be
SELECT r.id AS routesId, r.start_point AS startPointId, r.end_point AS endPointId, r.distance AS shortestDistance FROM routes r